src/CodeGen_Hexagon.cpp

```diff
@@ -2734,16 +2739,23 @@ void CodeGen_Hexagon::visit(const Load *op) {
             CodeGen_Posix::visit(op);
             return;
           }
-          int offset = b->value;
-          if (mod_rem.remainder != ((bytes_in_vector() + offset) % bytes_in_vector())) {
+          int offset = (b->value % lanes);
```

Comment:
If you use mod_imp instead of % then it'll always return a result between 0 and lanes-1.